buildscripts/Set-VersionInfo.ps1

```diff
@@ -0,0 +1,58 @@
+<#
+
+.SYNOPSIS
+
+Extracts build version information from Settings.proj and the
+environment. Then constructs several flavor version strings and
+uses them, along with the current year to
+
+  - update buildscripts/CommonAssemblyInfo.cs,
+  - update all project.json files, and
+  - emit TeamCity directives so it can know about the values
+
+#>
+
+[xml] $settings = Get-Content 'Settings.proj'
+$version_group = $settings.Project.PropertyGroup[1]
+
+$major = $version_group.Project_Major
+$minor = $version_group.Project_Minor
+$build = $version_group.Project_Build
+
+if ( $env:release_build -eq "true" )
+{
+    $version_suffix = ""
+}
+else
+{
+    $version_suffix = "-ci{0:00000}" -f [int]$env:build_number
+}
+
+$assembly_version = "$major.$minor.0";
```

Comment:
Can do. It seems we're dropping the build number from the `AssemblyFileInfo` as well (it was there before 55e152f49fc564581114fc0fa5908790c1bbc80d). I'll put that back too, if you've no objection, so it will be "M.m.p.b"